app/assets/javascripts/actions/story.js

```diff
@@ -26,24 +26,34 @@ export const editStory = (id, newAttributes) => ({
   newAttributes
 });
 
+export const setLoading = (id) => ({
+  type: actionTypes.SET_LOADING,
+  id
+});
+
 export const updateStory = (story, projectId) =>
   (dispatch, getState, { Story }) => {
     if (story._editing._isDirty) {
+      dispatch(setLoading(story.id))
       return Story.update(story._editing, projectId)
         .then((story) => {
           dispatch(updateStorySuccess(story));
-        });
+        })
+        .then(() => setLoading(story.id));
     }
     return dispatch(toggleStory(story.id));
   };
 
 export const deleteStory = (storyId, projectId) =>
-  (dispatch, getState, { Story }) =>
-    Story.deleteStory(storyId, projectId)
+  (dispatch, getState, { Story }) => {
+  dispatch(setLoading(storyId))
+    return Story.deleteStory(storyId, projectId)
       .then(() => dispatch(deleteStorySuccess(storyId)))
+      .then(() => dispatch(setLoading(storyId)))
       .catch(
         error => {
```

Comment:
What do u think about make this one line

```js
.catch(error => alert(error.message))
```

Reply:
good idea, but I actually made some more changes to the code and this catch is going to do some more stuff so I'll have to leave it open like that.